packages/fast-components-react-msft/src/button/button.tsx

```diff
@@ -42,21 +39,21 @@ class Button extends Foundation<IButtonHandledProps & IManagedClasses<IMSFTButto
     protected generateClassNames(): string {
         let classLocation: string;
 
-        if (this.props.primary) {
+        if (this.props.subtype === Subtype.primary) {
```

Comment:
I would prefer using a switch here now that this is isolated to one item we are checking against, we can also skip storing a variable, it can look like:

```
switch (this.props.appearance) {
    case Subtyle.outline:
        return super.generateClassNames(get(this.props, "managedClasses.button_outline"));
    // etc..
    default:
        return null;
}
```